plugin.video.diy/resources/lib/scraper.py

```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+# DIY Network Kodi Video Addon
+#
+from t1mlib import t1mAddon
+import json
+import re
+import os
+import datetime
+import urllib
+import xbmc
+import xbmcplugin
+import xbmcgui
+import HTMLParser
+import sys
+
+h = HTMLParser.HTMLParser()
+qp  = urllib.quote_plus
+uqp = urllib.unquote_plus
+UTF8     = 'utf-8'
+DIYBASE  = 'http://www.diynetwork.com%s'
+
+class myAddon(t1mAddon):
+
+  def getAddonMenu(self,url,ilist):
+     addonLanguage  = self.addon.getLocalizedString
+     html = self.getRequest('http://www.diynetwork.com/shows/full-episodes')
+     html = re.compile('<div class="capsule editorialPromo parbase section">(.+?)<div class="container-aside">', re.DOTALL).search(html).group(1)
+     a = re.compile('m-MediaBlock">.+?href="(.+?)".+?data-src="(.+?)".+?HeadlineText">(.+?)<.+?AssetInfo">(.+?)<', re.DOTALL).findall(html)
+     fanart = self.addonFanart
+     for i, (url, thumb, name, vidcnt) in list(enumerate(a[1:], start=1)):
+       name=name.strip()
+       vidcnt = vidcnt.strip()
+       html = self.getRequest(url.rsplit('/',1)[0])
+       plot = re.compile('"og:description" content="(.+?)"',re.DOTALL).search(html)
+       if plot is not None: plot = plot.group(1)
+       else: plot = name
+       infoList = {}
+       infoList['TVShowTitle'] = name
```

Comment:
That's really odd, the diy add-on is derived from the hgtv add-on and it's in that one. I'll clean this up tomorrow. Thanks.